### DISCUSSION TITLE
Stellarium crashing with AM5

### DISCUSSION THREAD

**Discussion by jairoamaral** [Stellarium/stellarium#3508] [Q&A] [answered]
I just bought an AM5 mount. Stellarium usually works wonderfully controlling it, but yesterday, in the last few hours of an imaging session, it began to consistently crash just a few minutes after I connect it to the mount.

Has anyone seen something similar? Do you know how should I proceed to find out what is causing this?

**Comment by gzotti** [comment]
No idea what AM5 is, but the logfile and extended logging available in the TelescopeControl plugin could help, together with hints given in the Release Notes. Is Stellarium the only application running while connected to the telescope?

**Reply by jairoamaral** [reply]
I give up.

I can't find any kind of logfile related to the TelescopeControl plugin. Not in Program Files. Not in Appdata.

Where it is, please?

**Reply by gzotti** [reply]
The TelescopeControl plugin settings panel, tab "Options", has a checkbox "Log telescope driver messages to files".  With this activated it presumably logs telescope driver messages to files `log_TelescopeServer<nr>.txt` in the user data directory (AppData/Roaming/Stellarium/). However, not currently using and not having developed the plugin I cannot say what these files include. 

Is there any volunteer out there who would want to improve chapter 16.2 of the User Guide in a cloudy night? Familiarity with telescope driving is required.

**Reply by jairoamaral** [reply]
I found this option and activated it. I still can't find any log_TelescopeServer* file.

**Reply by jairoamaral** [reply]
My search couldn't find anything like log_TelescopeServer* in my computer. However, I finally could identify and locate the relevant log file related to Stellarium.
[log crash 01.txt](https://github.com/Stellarium/stellarium/files/13468331/log.crash.01.txt)

**Reply by gzotti** [reply]
Finally it includes the first necessary bit of information: It is an ASCOM-driven telescope. 
When did problems occur? Did version 23.3 work for weeks, and only last week the same version started to crash, or did you only now update from 0.22.* to 23.3? 

Now, did you read the Release Notes? If you can help, keep this version and start bug hunting. If not, install Qt5-based 23.3.

**Reply by jairoamaral** [reply]
It began immediately, when I began using the AM5 mount with Stellarium. I bought it a few weeks ago.

It happened with version 23.2 and with 23.3 all the same.

I want to help. The crashes are not entirely random; they happen only when I'm switching or closing windows, so Stellarium becomes my active window and crashes. However, I also need this for work.  Is it possible to keep both without mutual interference? (Where is the Qt5-based 23.3?)

**Reply by gzotti** [reply, marked as answer]
You find Qt5-based 23.3 on the main stellarium.org site, right column, "all releases". And yes, this is another instance of #2874 or #2821.

**Reply by jairoamaral** [reply]
Using the Qt5-based 23.3 made the crashings disappear. I uset the AM5 mount an entire night with no crashes.

How do I help with the Qt6? Do I join a specific group and keep telling what happens when I use the Qt6 version?

**Reply by gzotti** [reply]
We have meanwhile understood that there is *some problem* on *some computers*. Further reports of "it crashed" won't bring us forward. 

Fixing this requires (apart from being able to reproduce it, being affected enough to become motivated and committed to the problem) compiling the program and running a debugger with enough knowledge how to do that and what to expect.  Once the core of the problem has been identified, fix the bad lines of code, compile, test again, and if it works, send us a "pull request". We will check whether this does not break the rest of the program, and likely accept the changes. We can also offer a €500 reward for successfully solving this problem.

**Comment by jairoamaral** [comment]
Hi! Thank you for the reply. AM5 is a telescope mount from ZWO. The Stellarium was the only application running, as far as I know.

I will look for the logfile.
